tutorials/radiant-mlhub-on-demand-training-data.ipynb

```diff
@@ -0,0 +1,1281 @@
+   "id": "443b5f79-3988-4d97-a6ef-58287341ee1b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "origin_label_items = search_stac_api(\n",
+    "    catalog_client=mlhub_catalog,\n",
+    "    collections=[BIGEARTHNET_LABEL_COLLECTION],\n",
+    "    bbox=LUXEMBOURG_AOI,\n",
+    "    datetime=BIGEARTHNET_TIME_RANGE,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "917c392c-9ae5-43e2-b7e0-71dd9f749cc2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "len(origin_label_items)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e9d121a9-e54b-4039-9cef-04277962c2ca",
+   "metadata": {},
+   "source": [
+    "This is another helper function that simply displays the geometry for labels from an ItemCollection overlayed on a map of the region."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2f203e31-5b09-4f2e-bf27-9a3ef8e3fc4d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def explore_search_extent(items: ItemCollection) -> None:\n",
+    "    \"\"\"Extracts geometry from ItemCollection to display polygons on a map.\n",
+    "\n",
+    "    Args:\n",
+    "        items: ItemCollection of Items retrieved from pystac_client search\n",
+    "\n",
+    "    Returns:\n",
+    "        GeoDataFrame object with the .explore() method called\n",
+    "    \"\"\"\n",
+    "    item_feature_collection = items.to_dict()\n",
+    "    geom_df = GeoDataFrame.from_features(item_feature_collection).set_crs(4326)\n",
+    "    return geom_df[[\"geometry\", \"datetime\"]].explore(\n",
+    "        column=\"datetime\", style_kwds={\"fillOpacity\": 0.2}, cmap=\"viridis\"\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ab923855-3c20-4f10-af00-d4175a54fdd4",
+   "metadata": {},
+   "source": [
+    "Here are the BigEarthNet chips with their bounding boxes that matched the spatial parameters for the city of Luxembourg and surrounding areas."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b86bf5d8-8dc8-491d-b3cd-1ea1263774ca",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "explore_search_extent(origin_label_items)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "13c6e607-c79b-4678-a483-9bacb0b3b1df",
+   "metadata": {},
+   "source": [
+    "### Download the entire label collection for BigEarthNet from Radiant MLHub"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2e131160-50bb-487f-8fcb-96d37ce80167",
+   "metadata": {},
+   "source": [
+    "We could certainly use the method above to query label Items directly from our connection to the Radiant MLHub API endpoint. However, on very large collections, such as in the case with BigEarthNet, pagination becomes a bottleneck issue in obtaining and resolving STAC items, as it only returns 100 items at a time.  Querying the entire Collection of nearly ~600,000 Items could take hours.\n",
```

Comment:
FWIW, the `limit` argument in pystac-client controls the size of the pages. But agreed that fetching 600,000 items through an API isn't what we should recommend.